### ISSUE TITLE
Sad Panda when trying to use "silence" on logger

### ISSUE THREAD

**Issue by evizitei** [instructure/paul_bunyan#1]
From Arc experience:

```
Ethan Vizitei [9:37 AM]  
so inst-jobs calls silence on the logger

[9:37]  
which before we use paul_bunyan goes through “LogSilencer"

[9:37]  
however, rails also defines “silence” on the Kernal

[9:38]  
and when we use paul_bunyan, it has no “silence” method, but the kernal does

[9:38]  
however the arguments don’t match

[9:38]  
(0 where expected 1)

[9:38]  
so it ​_does_​ fine a “silence” method, but it’s the one defined by “reporter.rb” in activesupport
```

Perhaps the answer is to have paul_bunyan forward calls to "silence" as well in it's delegatable methods?

**Comment by evizitei** [comment]
yeah, I don't know if it's a good idea, but I know that forwarding `silence` fixes this problem for our codebase:  

```
module PaulBunyan
  class LogRelayer
    extend Forwardable

    # delegate non-relayed methods to the primary logger
    DELEGATED_METHODS = %i(
      progname        progname=
      level           level=
      sev_threshold   sev_threshold=

      formatter       formatter=
      datetime_format datetime_format=

      close

      silence

      debug? info? warn? error? fatal?
    )
```

**Comment by miquella** [comment]
This would seem to break the purpose of silencing the logger as this would only silence the primary logger. All the other loggers would still receive any messages sent during this time.

I imagine proper support of `#silence` will need to come through calling the `#silence` method on all loggers supporting the interface.

**Comment by evizitei** [comment]
that makes sense to me

**Comment by tpickett66** [comment]
It looks like the definition of #silence can be found at https://github.com/rails/rails/blob/master/activesupport/lib/active_support/logger_silence.rb#L14 this could get interesting since it takes a block. I'll have to think about how we can yield to an arbitrary number of loggers.

**Comment by tpickett66** [comment]
Fixed by 5b4cbc1
